jupyterlab_automl/jupyterlab_automl/service.py

```diff
@@ -0,0 +1,283 @@
+    Tue = 2
+    Wed = 3
+    Thu = 4
+    Fri = 5
+    Sat = 6
+    Sun = 7
+
+
+class ChartInfo(Enum):
+    name = "name"
+    amount = "Number of Instances"
+
+
+class AutoMLService:
+    """Provides an authenticated AutoML Client and project info"""
+
+    _instance = None
+
+    def __init__(self):
+        self._client = automl_v1beta1.AutoMlClient()
+        self._parent = self._client.location_path(
+            AuthProvider.get().project, "us-central1"
+        )
+
+    @property
+    def client(self):
+        return self._client
+
+    @classmethod
+    def get(cls):
+        if not cls._instance:
+            cls._instance = AutoMLService()
+        return cls._instance
+
+    def get_models(self):
+        models = self._client.list_models(self._parent)
+        return {
+            "models": [
+                {
+                    "id": model.name,
+                    "displayName": model.display_name,
+                    "datasetId": model.dataset_id,
+                    "updateTime": model.update_time.ToMilliseconds(),
+                    "deploymentState": model.deployment_state,
+                    "metadata": "",
+                }
+                for model in models
+            ]
+        }
+
+    def get_datasets(self):
+        datasets = []
+        for dataset in self._client.list_datasets(self._parent):
+            dataset_type, metadata = self._get_dataset_metadata(dataset)
+            if dataset_type != DatasetType.other:
+                datasets.append(
+                    {
+                        "id": dataset.name,
+                        "displayName": dataset.display_name,
+                        "description": dataset.description,
+                        "createTime": dataset.create_time.ToMilliseconds(),
+                        "exampleCount": dataset.example_count,
+                        "datasetType": dataset_type.value,
+                        "metadata": metadata,
+                    }
+                )
+        return {"datasets": datasets}
+
+    def _get_dataset_metadata(self, dataset):
+        dataset_type = parse_dataset_type(dataset)
+        if dataset_type == DatasetType.TBL:
+            metadata = {
+                "primary_table_spec_id": dataset.tables_dataset_metadata.primary_table_spec_id,
+                "target_column_spec_id": dataset.tables_dataset_metadata.target_column_spec_id,
+                "weight_column_spec_id": dataset.tables_dataset_metadata.weight_column_spec_id,
+                "ml_use_column_spec_id": dataset.tables_dataset_metadata.ml_use_column_spec_id,
+                "stats_update_time": dataset.tables_dataset_metadata.stats_update_time.ToMilliseconds(),
+            }
+        elif dataset_type == DatasetType.ICN:
+            metadata = {
+                "classification_type": dataset.image_classification_dataset_metadata.classification_type,
+            }
+        elif dataset_type == DatasetType.IOD:
+            metadata = ""
```

Comment:
should there be an else here if the dataset is not one of the three types? or you could define metadata outside of the if/else clause.

Reply:
+1 will address this